[Quality] unused parameter 'output_index_' [-Werror,-Wunused-parameter]

https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/7494c1c611382069243709cd9eab0efb90843c7c/include/ck/tensor_operation/gpu/grid/gridwise_sparse_embeddings_forward_layernorm.hpp#L188

The above line is causing issues with compiler included in ROCm 5.4.2:
```
In file included from /home/junliu/composable_kernel/include/ck/tensor_operation/gpu/device/impl/device_sparse_embeddings_forward_layernorm.hpp:15:
/home/junliu/composable_kernel/include/ck/tensor_operation/gpu/grid/gridwise_sparse_embeddings_forward_layernorm.hpp:188:34: error: unused parameter 'output_index_' [-Werror,-Wunused-parameter]
                        [&](auto output_index_) -> auto& {
                                 ^
1 error generated when compiling for gfx908.
```
not sure how it passed CK CI tests. Do we have a static stage checking via cppcheck or clang-tidy?

--- a/include/ck/tensor_operation/gpu/grid/gridwise_sparse_embeddings_forward_layernorm.hpp
+++ b/include/ck/tensor_operation/gpu/grid/gridwise_sparse_embeddings_forward_layernorm.hpp
@@ -185,7 +185,7 @@ struct GridwiseSparseEmbeddingsForwardLayernorm
                         },
                         Number<NumEmbeddings>{});
                     auto out_data_refs = generate_tie(
-                        [&](auto output_index_) -> auto& {
+                        [&](auto) -> auto& {
                             return acc_thread_buf(Number<register_offset>{});
                         },
                         Number<1>{});

--- a/include/ck/tensor_operation/gpu/grid/gridwise_sparse_embeddings_forward_layernorm.hpp
+++ b/include/ck/tensor_operation/gpu/grid/gridwise_sparse_embeddings_forward_layernorm.hpp
@@ -185,9 +185,7 @@ struct GridwiseSparseEmbeddingsForwardLayernorm
                         },
                         Number<NumEmbeddings>{});
                     auto out_data_refs = generate_tie(
-                        [&](auto) -> auto& {
-                            return acc_thread_buf(Number<register_offset>{});
-                        },
+                        [&](auto) -> auto& { return acc_thread_buf(Number<register_offset>{}); },
                         Number<1>{});
                     unpack2(emb_elementwise_op, out_data_refs, in_data_refs);
                 });

